File: spark-java/src/main/java/adj/spark/core/action/CountOperation.java
package adj.spark.core.action;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

/**
 * count: 统计元素个数
 * @author adolf felix
 *
 */
public class CountOperation {
	public static void main(String[] args) {
		SparkConf conf = new SparkConf();
				  conf.setAppName("CollectOperation")
				      .setMaster("local");
		@SuppressWarnings("resource")
		JavaSparkContext jsc = new JavaSparkContext(conf);
		
		List<Integer> numberList = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9);
		JavaRDD<Integer> numberRDD = jsc.parallelize(numberList);
		
		Long count = numberRDD.count();
		System.out.println("count = " + count);
	}
}

File: spark-scala-maven/src/main/scala/adj/spark/core/action/CountOperation.scala
package adj.spark.core.action

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object CountOperation {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
               .setMaster("local")
               .setAppName("CountOperation")
    val sc = new SparkContext(conf)
    val numberArr = Array(1, 2, 3, 4, 5, 6, 7, 8, 9)
    
    val numberRDD = sc.parallelize(numberArr, 3);
    val douleRDD = numberRDD.map(number => number * 2)
    
    val count = douleRDD.count()
    println("count = " + count)
  }
}